src/main.rs

```diff
@@ -100,13 +99,14 @@ struct Init {
     /// The path at which to create the config file.
     #[arg(default_value = ".daft.toml")]
     path: PathBuf,
+
+    /// The provider to use - either 'aws' (default) to auto-generate a cluster or 'k8s' for existing Kubernetes clusters
+    #[arg(long, default_value = "aws")]
+    provider: String,
```

Comment:
We should remove this, since we shouldn't have both `[setup.<PROVIDER>]` and `provider = "<PROVIDER>"`.

However, two other points as well:
1. We tend to use `Arc<str>` instead of `String` and `Arc<Path>` instead of `PathBuf` (in Daft and in Daft-Launcher).
2. If you have a fixed list of string variants that you want to deserialize (e.g., "aws", "gcp", "k8s"), it's better to represent those as a new enum instead of a `String`. I.e.:

```rs
#[derive(Debug, Deserialize, Clone, Copy, PartialEq, Eq)]
enum Provider { Aws, Gcp, K8s }
```

This is because `provider: String` will allow any arbitrary string to be accepted, but `provider: Provider` will only accept "aws", "gcp", and "k8s" (and any other value will throw a very nice error message).